mongodb_store/scripts/mongodb_server.py

```diff
@@ -38,6 +38,7 @@ def __init__(self):
 
         self.test_mode = rospy.get_param("~test_mode", False)
         self.repl_set = rospy.get_param("~repl_set", None)
+        self.bind_to_host = rospy.get_param("~bind_to_host", False)
```

Comment:
Only thing I'm a little concerned about (and I may be wrong here) is that `self.bind_to_host` is used either a Bool or a String here. In particular, with the launch files that set it to `default="false"` without explicitly giving a type here, how is it assured this is seen as a boolean `False` and not as a string `"false"` (i.e. an ip address/hostname). Maybe it should be changed to always be a string, that, if empty, the default is to bind all IPs?

Reply:
I designed this feature as a boolean flag to prevent this from being a breaking change. If the value is True (or event a string, any string actually) it will pass the existing host value (self._mongo_host) to the mongod process and it will only bind to that. I deliberately did not want to bring another string variable with the host to actually bind to since you may end up in a situation where `self._mongodb_host` says `localhost`, but the server is actually bound to something else. Using a boolean flag makes the consistent.

The parameter is designed to be passed to the launch file as an argument, sadly the argument does not have a type attribute. However, the parameter attribute does. Would you like me to update line 38 to specify the type? Like this:
```<param name="bind_to_host" type="bool" value="$(arg ..)"/>```
this would catch anyone using strings.

Reply:
The argument / parameter is optional. If _not_ set it defaults to the current behavior which is to bind to all interfaces.

When set, it will bind to the address specified by `HOSTNAME`, which is already exposed as the parameter `/mongodb_host`. `HOSTNAME` defaults to `localhost`, which means that it will only bind to localhost, preventing anyone on the network from accessing it, which is the functionality I'm trying to achieve.

Reply:
My apologies, I misunderstood. All is good. `self.bind_to_host` is always a bool. I somehow read into it that `self.bind_to_host` defines the actual host/ip to bind to. But that's not the case. All good!

Reply:
No problem. Happy to solve any misunderstandings. Thanks for taking the time to evaluate my work.